Configuration/TCA/tx_dlf_solrcores.php

```diff
@@ -49,7 +49,6 @@
                 'max' => 255,
                 'eval' => 'alphanum,nospace,required,unique',
                 'default' => '',
-                'readOnly' => true,
```

Comment:
This allows changing `index_name` (i.e. the core name) in production, which is quite dangerous. Is this really necessary for the scenario you are trying to achieve? As far as I understand you can get all your described scenarios to work by just allowing to add cores without a working Solr connection (i. e. the changes in `DataHandler.php` above). There is no need to allow changes of already configured core names. Or am I missing something?

I mean, this works for me. I am just concerned about the removed protection against potentially dangerous changes.

Reply:
Of course this is dangerous. Or let's better say "stupid", if you change the index_name and no corresponding Solr core exists. But it doesn't unload/remove the Solr core or something else happens. You only get a red warning, that the current configuration doesn't reach a Solr core. So the risk is low.

When you need to change the `index_name`? For example 

- if you move to a new TYPO3-server, export your tree with the impexp module. All records are created on import. So you create a Solr core, too. And cannot "switch" to the right core without SQL-commands.
- if you have a read-only Solr core like the zeitungsportal one, you need to set the given `index_name` 

This is how it looked like before:

![New TYPO3 site  TYPO3 CMS 9 5 30 ](https://user-images.githubusercontent.com/3354500/130061333-ac590e68-1a2e-4b11-8824-f2cd4fde902b.png)

And this happens, if someone enters a wrong `index_name`:

![New TYPO3 site  TYPO3 CMS 9 5 30  (1)](https://user-images.githubusercontent.com/3354500/130062831-0bef3d27-b034-4459-972a-23b39257a5d8.png)

Reply:
You're right! I didn't think this through all the way. 😄